web/src/main/java/com/bazaarvoice/emodb/web/auth/OwnerDatabusAuthorizer.java

```diff
@@ -0,0 +1,237 @@
+     * typically temporally clustered by table.  Instantiation the read permission for a table is relatively inexpensive
+     * but again for the sake of efficiency during databus fanout should be minimized.  Unlike with the previous cache,
+     * however, the permission instance is not tied to any particular user and can be cached indefinitely with no
+     * negative efffects.  For this reason the permissions are cached separately to reduce frequent permission resolution.
+     */
+    private final LoadingCache<String, Permission> _readPermissionCache;
+
+    private final PermissionResolver _permissionResolver;
+
+    @Inject
+    public OwnerDatabusAuthorizer(InternalAuthorizer internalAuthorizer, final PermissionResolver permissionResolver,
+                                  MetricRegistry metricRegistry) {
+        this(internalAuthorizer, permissionResolver, metricRegistry, DEFAULT_PERMISSION_CHECK_CACHE_SIZE,
+                DEFAULT_PERMISSION_CHECK_CACHE_TIMEOUT, DEFAULT_READ_PERMISSION_CACHE_SIZE);
+    }
+
+    public OwnerDatabusAuthorizer(InternalAuthorizer internalAuthorizer, final PermissionResolver permissionResolver,
+                                  MetricRegistry metricRegistry, int permissionCheckCacheSize,
+                                  Duration permissionCheckCacheTimeout, int readPermissionCacheSize) {
+        _internalAuthorizer = checkNotNull(internalAuthorizer, "internalAuthorizer");
+        _permissionResolver = checkNotNull(permissionResolver, "permissionResolver");
+
+        if (permissionCheckCacheSize > 0) {
+            checkNotNull(permissionCheckCacheTimeout, "permissionCheckCacheTimeout");
+            checkArgument(!permissionCheckCacheTimeout.isLongerThan(MAX_PERMISSION_CHECK_CACHE_TIMEOUT),
+                    "Permission check cache timeout is too long");
+
+            _permissionCheckCache = CacheBuilder.newBuilder()
+                    .maximumSize(permissionCheckCacheSize)
+                    .expireAfterWrite(permissionCheckCacheTimeout.getMillis(), TimeUnit.MILLISECONDS)
+                    .recordStats()
+                    .build(new CacheLoader<OwnerTableCacheKey, Boolean>() {
+                        @Override
+                        public Boolean load(OwnerTableCacheKey key) throws Exception {
+                            return ownerCanReadTable(key._ownerId, key._table);
+                        }
+                    });
+
+            if (metricRegistry != null) {
+                // Getting the full benefits of permission check caching requires tuning.  Publish statistics to
+                // give visibility into performance.
+                metricRegistry.register(MetricRegistry.name("bv.emodb.databus", "authorizer", "read-permission-cache", "hits"),
+                        new Gauge<Long>() {
+                            @Override
+                            public Long getValue() {
+                                return _permissionCheckCache.stats().hitCount();
```

Comment:
That's why I exposed "hitCount" and "missCount".  I didn't want to just expose "hitRate" since that hides the actual number of cache lookups that are performed.